app/views/sign_up/recovery_codes/_recovery_page_controller.js.erb

```diff
@@ -0,0 +1,96 @@
+var modalSelector = '#personal-key-confirm';
+var modal = new LoginGov.Modal({el: modalSelector });
+
+var recoveryWords = [].slice.call(document.querySelectorAll('[data-recovery]'));
+var formEl = document.getElementById('confirm-key');
+var inputs = [].slice.call(formEl.elements).filter(function(el) { return el.type === 'text'});
+var modalTrigger = document.querySelector('[data-toggle="modal"]');
+var modalDismiss = document.querySelector('[data-dismiss="personal-key-confirm"]')
+
+var reminderEl = document.getElementById('recovery-code-reminder');
+
+var isInvalidForm = false;
+var alertFallback = '<div class="alert alert-alert" role="alert" id="recovery-code-alert">' +
+                    "<%= t('users.recovery_code.confirmation_error') %>" +
```

Comment:
Yeah I just sort of threw this on here, we can also just construct it entirely in JS

`document.createElement('div')`, `div.className = 'classes'` etc

Which approach would you expect to see in a rails app?